etc/bash_completion.d/yab

```diff
@@ -0,0 +1,16 @@
+# bash completion for yab executable
+have yab &&
+_yab()
+{
+    # All arguments except the first one
+    args=("${COMP_WORDS[@]:1:$COMP_CWORD}")
+
+    # Only split on newlines
+    local IFS=$'\n'
+
+    # Call completion (note that the first element of COMP_WORDS is
+    # the executable itself)
+    COMPREPLY=($(GO_FLAGS_COMPLETION=1 ${COMP_WORDS[0]} "${args[@]}"))
+    return 0
+} &&
```

Comment:
assume `&&` here is to make sure we only register the function if `yab` is found?

can we instead exit early if it's not found and avoid the `&&`?

Reply:
Are you really sure you want this?

1. This is the style of bash-completion. Should we break from that?
2. This is shell we are talking about. `&&` uses the status code not what the function prints (which is what it actually returns). The check will be longer and not buy anything:
```
if [ $? -eq 0 ]
then
    # register autocomplete
else
    # do nothing
fi
```
3. We cannot `exit`. This whole script is executing by sourcing - within the invoking shell rather than in a subshell. An `exit` will close the shell and prevent you from running. If you mean _`return`_ - again, it will be uglier

Reply:
If this is the style that bash completions use, then it seems fine.